File: src/main/java/xyz/mlhmz/savingscategorization/models/Transaction.java
package xyz.mlhmz.savingscategorization.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "transactions")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Transaction {
    @Id
    String id;

    @Column(columnDefinition = "TEXT")
    String reason;

    String issuer;

    LocalDate date;

    double amount;

    @ManyToOne
    @JoinTable(name = "pocket_transactions", joinColumns = {@JoinColumn(name = "transactions_id", referencedColumnName = "id")},
        inverseJoinColumns = {@JoinColumn(name = "pocket_uuid", referencedColumnName = "uuid")})
    private Pocket pocket;
}

File: frontend/src/types/Transaction.ts
import { z } from "zod";

export interface Transaction {
  id?: string;
  reason?: string;
  issuer?: string;
  date?: string;
  amount?: number;
  pocket?: TransactionPocket;
  hideForced?: boolean;
  pocketForced?: boolean;
  forceReason?: string;
}

export const TransactionMutation = z.object({
  forceReason: z.string().optional(),
  pocketUuid: z.string().uuid().optional(),
  pocketForced: z.boolean().optional(),
  hideForced: z.boolean().optional(),
});

export type TransactionMutation = z.infer<typeof TransactionMutation>;

export type Transactions = Array<Transaction>;

export interface TransactionPocket {
  uuid?: string;
  name?: string;
}
